xocto/storage/storage.py

```diff
@@ -1179,6 +1184,10 @@ def get_size_in_bytes(self, *, s3_object: S3Object) -> int:
         file_stats = os.stat(filepath)
         return file_stats.st_size
 
+    def get_last_modified(self, *, s3_object: S3Object) -> datetime.datetime:
```

Comment:
Because you are in local file system, I don't think this will work.  Your implementation in the prior commit should work.
```
    def get_last_modified(self, *, s3_object: S3Object) -> datetime.datetime:
        filepath = os.path.join(self.storage_root, s3_object.bucket_name, s3_object.key)
        file_stats = os.stat(filepath)
        return datetime.datetime.fromtimestamp(file_stats.st_mtime)
```